runtime/interpreter/errors.go

```diff
@@ -685,9 +688,9 @@ type ResourceConstructionError struct {
 	LocationRange
 }
 
-var _ errors.UserError = ResourceConstructionError{}
+var _ errors.InternalError = ResourceConstructionError{}
 
-func (ResourceConstructionError) IsUserError() {}
+func (ResourceConstructionError) IsInternalError() {}
```

Comment:
This is not an internal error, but is reported e.g. when a user tries to call the resource constructor function outside of the declaration location.

Maybe add a comment to the error type declaration instead which clarifies this.

Reply:
This is a static error isn't it? 

IIRC the runtime version is more of a defensive check against any potential implementation errors we may have made in the static check. The only test that we have for this error requires creating a function value in Go that cannot be constructed in Cadence; if that's the only way we have to trigger an error it seems to me that it shouldn't be considered a user error since there is no way for a user to cause it.

Reply:
yeah, I think this is just the defensive check for the interpreter. Ideally should be prevented at the checker, so this is OK to be an internal error.